docs/deferrables.rst

```diff
@@ -1,23 +1,24 @@
-===============================================
-Understanding the Mechanism of Sanity Functions
-===============================================
+===================================================
+Understanding the Mechanism of Deferrable Functions
+===================================================
 
-This section describes the mechanism behind the sanity functions that are used for the sanity and performance checking.
-Generally, writing a new sanity function is as straightforward as decorating a simple Python function with the :func:`reframe.utility.sanity.sanity_function` decorator.
-However, it is important to understand how and when a deferrable function is evaluated, especially if your function takes as arguments the results of other deferrable functions.
+This section describes the mechanism behind deferrable functions, which in ReFrame, they are mostly used for sanity and performance checking.
```

Comment:
Do you mean the "mostly" in this sentence? I don't have a strong opinion on it, so you decide :) I just went for the more general description here to avoid locking the way reframe does things currently with the docs. One could for example do sanity without using deferrable functions, and if we take that "mostly" out, it feels like you must use deferrables for it. In any case, I'm fine with either.